### PR TITLE
Windows: add docker development documentation

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Windows: add docker development documentation
Additional Description:
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @wrowe @nigriMSFT @davinci26 @lizan for any feedback/edits

File: ci/README.md [hunk 1]
```diff
@@ -18,18 +20,26 @@ one with an Envoy binary with debug (`envoyproxy/envoy-alpine-debug`) symbols an
...
 * `envoyproxy/envoy-build` &mdash; alias to `envoyproxy/envoy-build-ubuntu`.
 * `envoyproxy/envoy-build-ubuntu` &mdash; based on Ubuntu 18.04 (Bionic) with GCC 9 and Clang 9 compiler.
 * `envoyproxy/envoy-build-centos` &mdash; based on CentOS 7 with GCC 9 and Clang 9 compiler, this image is experimental and not well tested.
+* `envoyproxy/envoy-build-windows2019` &mdash; based on Windows 2019 LTS with VS 2019 Build Tools.
 
 The source for these images is located in the [envoyproxy/envoy-build-tools](https://github.com/envoyproxy/envoy-build-tools)
 repository.
 
-We use the Clang compiler for all CI runs with tests. We have an additional CI run with GCC which builds binary only.
+We use the Clang compiler for all Linux CI runs with tests. We have an additional Linux CI run with GCC which builds binary only.
```
Comment (lizan) [inline ci/README.md:42]: We run Windows CI builds binary too?

File: ci/README.md [hunk 2]
```diff
@@ -18,18 +20,26 @@ one with an Envoy binary with debug (`envoyproxy/envoy-alpine-debug`) symbols an
 Both images are pushed with two different tags: `<hash>` and `latest`. Parallel to the Ubuntu images above, `<hash>` corresponds to the
 master commit at which the binary was compiled, and `latest` corresponds to a binary built from the latest tip of master that passed tests.
 
+## Windows 2019 Envoy image
+
+The Windows 2019 based Envoy Docker image at [`envoyproxy/envoy-build-windows2019:<hash>`](https://hub.docker.com/r/envoyproxy/envoy-build-windows2019/)
+is used for CI checks, where `<hash>` is specified in [`envoy_build_sha.sh`](https://github.com/envoyproxy/envoy/blob/master/ci/envoy_build_sha.sh).
+Developers may work with `envoyproxy/envoy-build-windows2019:latest` to provide a self-contained environment for building Envoy binaries and
```
Comment (lizan) [inline ci/README.md:27]: I don't think we tag `latest`.

Comment (sunjayBhatia) [reply ci/README.md:27]: Good catch, will remove that line

File: ci/README.md (hunk 1 above)
Comment (sunjayBhatia) [reply ci/README.md:42]: Ah, I omitted that b/c we are intending to build the binary and enable running tests shortly and didn't want to have too much churn when that state changes; will adjust this to include some info about the current state Windows

**Pull request merged**